### ISSUE TITLE
v1 bug: Panic when printing completions

### ISSUE THREAD

**Issue by VirrageS** [urfave/cli#1093]
## my urfave/cli version is

v1.22.2, v1.22.3

## Checklist

* [x] Are you running the latest v1 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v1 manual is [here](https://github.com/urfave/cli/blob/master/docs/v1/manual.md)
* [x] Did you perform a search about this problem? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## Dependency Management

- [x] My project is using go modules.
- [ ] My project is using vendoring.
- [ ] My project is automatically downloading the latest version.
- [ ] I am unsure of what my dependency management setup is.

## Describe the bug

I'm experiencing panic when trying to print bash completions for partially completed flag  `cmd subcmd --abc[TAB]`.

## To reproduce

```go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/urfave/cli"
)

func main() {
	app := cli.NewApp()
	app.Name = "example"
	app.EnableBashCompletion = true
	app.Commands = []cli.Command{{
		Name:   "cmd",
		Action: func(c *cli.Context) error { return nil },
		Flags:  []cli.Flag{cli.StringFlag{Name: "abc1"}, cli.StringFlag{Name: "abc2"}},
		BashComplete: func(c *cli.Context) {
			fmt.Println(c.NArg())
			fmt.Println("--abc1")
			fmt.Println("--abc2")
		},
	}}
	app.Action = func(c *cli.Context) error {
		fmt.Println("Hello friend!")
		return nil
	}

	err := app.Run(os.Args)
	if err != nil {
		log.Fatal(err)
	}
}
```

1. Build code above
2. Load completions
3. `example cmd --abc[TAB]`

## Observed behavior

Panic

## Expected behavior

No panic. I also would expect that I will get completions for the flag:
```console
$ example cmd --abc[TAB]
--abc1 --abc2
```

## Additional context

https://github.com/urfave/cli/blob/c354ceca3e45cda773c3846bcb0c67cec4046b03/command.go#L117-L125

Code here looks very suspicious. If we have `err` returned in `c.parseFlags` then `set` will be `nil` and `checkCommandCompletions` will access `ctx` which has `ctx.flagSet == nil`. This means that any calls inside the `checkCommandCompletions` accessing `flagSet` will panic (like `.NArg()`).

There was also similar issue for the `v2`: https://github.com/urfave/cli/issues/944. It would be nice to have that fix backported to the `v1` as well.

## Run `go version` and paste its output here

```
go version go1.14 darwin/amd64
```

**Comment by lynncyrin** [comment]
> It would be nice to have that fix backported to the v1 as well.

@VirrageS PRs are welcome, if you want to take care of this yourself! The v1 code lives here => https://github.com/urfave/cli/tree/v1

**Comment by lynncyrin** [comment]
released in 1.22.4! ✅ https://github.com/urfave/cli/releases/tag/v1.22.4
